Client/Coordinators/Router/RouteBuilder.swift

```diff
@@ -4,12 +4,16 @@
 
 import Foundation
 import CoreSpotlight
+import Shared
 
-struct RouteBuilder {
-    private var isPrivate: () -> Bool
+final class RouteBuilder {
+    private var isPrivate: Bool?
```

Comment:
Optional Booleans look a bit off. There's `true`, `false` and missing? Do we need the missing case?